lib/bootsnap/load_path_cache/path_scanner.rb

```diff
@@ -34,18 +26,18 @@ def self.call(path)
         dirs = []
         requirables = []
 
-        Dir.glob(path + REQUIRABLES_AND_DIRS).each do |absolute_path|
+        Dir.glob(path + ALL_FILES).each do |absolute_path|
           next if contains_bundle_path && absolute_path.start_with?(BUNDLE_PATH)
-          relative_path = absolute_path.slice!(relative_slice)
+          relative_path = absolute_path.slice(relative_slice)
 
-          if relative_path.end_with?('/')
-            dirs << relative_path[0..-2]
-          else
+          if File.directory?(absolute_path)
+            dirs << relative_path
+          elsif REQUIRABLE_EXTENSIONS.include?(File.extname(relative_path))
             requirables << relative_path
           end
         end
 
-        [requirables, dirs]
+        [requirables.uniq, dirs.uniq]
```

Comment:
Im a bit concerned by the `uniq` as I assume a unique function on an array would be slow.
Otherwise, the rest of the changes seem to make sense to me.

I'd like @burke to look as well though

Reply:
FWIW `Array#uniq` is O(n) (and not O(n^2)) as it just uses a Hash internally to deduplicate.
So I'd guess it's not a bottleneck with a Dir.glob call around.